x/gamm/keeper/msg_server.go

```diff
@@ -26,16 +27,28 @@ func NewBalancerMsgServerImpl(keeper *Keeper) balancer.MsgServer {
 	}
 }
 
+func NewStableswapMsgServerImpl(keeper *Keeper) stableswap.MsgServer {
+	return &msgServer{
+		keeper: keeper,
+	}
+}
+
 var (
-	_ types.MsgServer    = msgServer{}
-	_ balancer.MsgServer = msgServer{}
+	_ types.MsgServer      = msgServer{}
+	_ balancer.MsgServer   = msgServer{}
+	_ stableswap.MsgServer = msgServer{}
```

Comment:
Why this change? The msgServer{} struct satisfies the relevant interfaces?

Reply:
This is to avoid the allocation of `msgServer{}` by creating a struct. The suggested pattern is used to ensure that the struct implements the interface but doesn't do the allocation

Reply:
This doesn't even get compiled though, it's just a compile time type check. I don't think we should worry about optimizing compile times

Reply:
Golangs dead code elimination should prevent this from entering the binary

Reply:
Yeah, I was wrongly assuming that only this pattern is right. I looked at: https://go.dev/doc/faq#guarantee_satisfies_interface which also suggests both. Thanks for pointing this out